source/server/lsp-servers.ads

```diff
@@ -125,6 +125,14 @@ package LSP.Servers is
    --  notification/request it's currently processing. This should only be
    --  called from the processing task.
 
+   --------------------------
+   -- Cancellation support --
+   --------------------------
+
+   function Cancel_Current (Self : Server) return Boolean;
```

Comment:
I'm afraid making server to have just one _current_ request could be too restrictive for us. In 3.15 we have _partial result_, so server can execute several request at once. Consider long running "find all references" and working tooltips during its execution. It's very attractive to have.

Reply:
no, it will never be possible to have the *processing* task execute more than one current request: LAL is not task safe

Reply:
We could do this way

1. take 10 files from Source_Files and execute 'find references' for them
2. send partial results and progress report for these 10 files
3. check if we have 'over' or similar request and execute it if any
4. continue with next 10 files, etc

Reply:
yes, that could be interesting! will go back to the drawing board.